### PR TITLE
Touch system.yaml to trigger cache update

### PR DESCRIPTION
**Pull request by leotiger**
This is related with issue #2093.

Background for this PR:
A site that is configured to use cache but disallowing detection of changes to obtain the maximum speed, faces some drawbacks:

* if using and working with admin plugin changing content implies clearing the caches manually
* if a site implements automated updates, e.g. within a store implementation handling the update of stocks, this won't show up till the next manual cache clearance
* if the site uses a volatile memory cache, e.g. memcached or apcu, a selective or partial cache update won't clear the volatile cache

Solution:
This pull request proposal tries to solve these problems adding

* a configuration option to the system configuration called autotouch
* adds a call to the new autotouch function in the cache class inside of the page class in the save function
* adds a parameter option to the clearCache method which allows to add the new "touch-only" option in the admin plugin interface or to call this in code

Update:
To allow better support for bulk updates and to avoid touching system.yaml with every save / change plugins and other code implementations can now emit a ``$this->grav['cache']->disableAutotouch();`` before starting with bulk operations and once finalized they can add two lines of code to trigger the behavior:

```
        $user_config = USER_DIR . 'config/system.yaml';
        touch($user_config);
```
(that's better and failsave this way as we don't need an additional function to activate autotouch again. As this is developers only, probably the best way to fix this all...)

P.S.: And optimized as a 4-liner:

```
$user_config = USER_DIR . 'config/system.yaml';
if (file_exists($user_config)) {
    touch($user_config);
}
```

### CODE REVIEW

**Comment by leotiger** [comment]
Probably, to improve further, it would be good to add an additional parametrization, a runtime option that allows to handle this in code better:

* switch off automatic handling for saves as every single save triggers the refresh
* touch system.yaml in code implementations once all bulk changes have been executed

**Comment by leotiger** [comment]
I've prepared the necessary changes for the admin plugin as well. Please request if this gets approved.

**Comment by leotiger** [comment]
Sidenote: This implementation (or its impact) is one of the key optimizations, allowing me to obtain a 100 out of 100 in Google Page Speed Insights for a highly graphical and dynamic GRAV website while still being able to offer an online shop with all the necessary underlying updates... they trigger on demand, on sale this way...

**Comment by leotiger** [comment]
Nonce emits onFormValidationError, so this is easy as well.

File: system/src/Grav/Common/Cache.php [hunk 1]
```diff
@@ -422,24 +429,55 @@ public static function clearCache($remove = 'standard')
...
+    }
+    
+    /**
+     * Helper that allows to re-fill caches when cache.method is set to none for speed
+     * and volatile caches like memcached or apcu are used
+     * 
+     */
+    public function autotouch()
+    {
+        $user_config = USER_DIR . 'config/system.yaml';
```
[the remark this answers is not in the document]
Comment (leotiger) [reply system/src/Grav/Common/Cache.php]: @mahagr I've updated cache.php using the helper function findresource.

**Comment by mahagr** [comment]
`bin/grav clear --invalidate` and `Cache::clearCache('invalidate')` have been added a while ago and it does basically the same as this pull request.

**Pull request closed without merging**